pkg/commands/workload_apply.go

```diff
@@ -112,9 +129,23 @@ func (opts *WorkloadApplyOptions) Exec(ctx context.Context, c *cli.Config) error
 
 		workload.Spec.MergeServiceAccountName(serviceAccountCopy)
 	}
+	if opts.UpdateStrategy == MergeUpdateStrategy {
+		workload.Merge(fileWorkload)
+	}
 
-	workload.Merge(fileWorkload)
+	if opts.UpdateStrategy == ReplaceUpdateStrategy {
```

Comment:
When trying to assign 
`workload = fileWorkload`

all the object metadata got lost so it threw an error for some fields that the system automatically populates. So this approach tries to first assign the metadata that comes from the workload that's currently in the cluster and then replace `labels` and `annotations` according to the user changes in the file.

Reply:
Some other way that I tried and works as well is trying to just add the system populated fields to the workload metadata after it has been replaced with the workload from file

```
        workload = fileWorkload
        if currentWorkload != nil {
            if !reflect.DeepEqual(currentWorkload.ObjectMeta, (metav1.ObjectMeta{})) {
                wldMeta := workload.GetObjectMeta()
                currentWldMeta := currentWorkload.GetObjectMeta()

                wldMeta.SetResourceVersion(currentWldMeta.GetResourceVersion())
                wldMeta.SetUID(currentWldMeta.GetUID())
                wldMeta.SetGeneration(currentWldMeta.GetGeneration())
            }
        }
```

I didn't use this approach because, if eventually, there are other `n` fields like this, then we'll have to manually add them and it will become untenable in time

Reply:
On the other hand, if user tries to update fileds like `OwnerReferences`, `Finalizers`, `ManageFields`, then these won't be updated, 'cause some logic to manage these is missing.

Reply:
As per discussion with @shashwathi, the decided approach is to copy all of the metadata as it comes and then set the system populated fields, since metadata is not likely to have big/non-manageable changes.